### ISSUE TITLE
Wrong version when building outside of ejabberd git (i.e., ejabberd releases)

### ISSUE THREAD

**Issue by joudinet** [processone/ejabberd#658]
Hi,
Whenever I download an ejabberd release (i.e., a tarball), I get the wrong version name because the configure.ac looks for the output of `git describe --tags`. Do you think it could be possible for you to hardcode the version number in `AC_INIT` for every release?

**Comment by cromain** [comment]
in order to adapt every release, we let autoconf.ac read version from git. as we prefer avoid commit a fixed version change at every release, we include generated configure in official source tarball, with static versions included:
```
 580 # Identity of this package.
 581 PACKAGE_NAME='ejabberd'
 582 PACKAGE_TARNAME='ejabberd'
 583 PACKAGE_VERSION='15.06'
 584 PACKAGE_STRING='ejabberd 15.06'
 585 PACKAGE_BUGREPORT='ejabberd@process-one.net'
```
but, as you more likely need to run autoconf, the simplest solution i suggest is to run sed on configure.ac before calling autogen.sh. something like:
    sed -i -e 's/echo 0\.0/echo 15\.06/' configure.ac

**Comment by cromain** [comment]
maybe be can apply this change on the fly on configure.ac when generating official source tarball by the way

**Comment by cromain** [comment]
good news, i could apply change to make configure.ac using tag version when generating source tarball.
so this issue will be permanently fixed from next release.

**Comment by joudinet** [comment]
Thanks @cromain.
Hope you change the whole git describe command when generating the source tarball and not only the `echo 0.0`. Otherwise, downloading the source tarball to a folder that belongs to another git repository will end up using a tag from this git repository instead of ejabberd correct version.
That's actually what happens to me when buildroot downloads ejabberd sources: ejabberd version is a buildroot tag instead of 0.0.

**Comment by cromain** [comment]
actually i just changed the echo as we need the git describe to generate version on the fly.
i did not thought about the buildroot case, you're right it's best to just generate tarball without the git stuff, i'll check how to improve this without breaking my own build process :)

**Comment by cromain** [comment]
i've permanently fix the glitch. now configure.ac included in source tarball have this line, with $vsn set to release version. It will be available in 15.07.
AC_INIT(ejabberd, $vsn, ejabberd@process-one.net, ejabberd)

**Comment by joudinet** [comment]
Thanks @cromain.
I've just downloaded ejabberd-15.07.tar.gz from ejabberd.im and saw your changes. :+1:
